rpc/v10.ts

```diff
@@ -1 +1,2606 @@
+	/**
+	 * Channel id to join (or `null` to leave)
+	 */
+	channel_id: Snowflake | null;
+	/**
+	 * Asynchronously join channel with time to wait before timing out
+	 */
+	timeout?: number;
+	/**
+	 * Forces a user to join a voice channel
+	 */
+	force?: boolean;
+	/**
+	 * After joining the voice channel, navigate to it in the client
+	 */
+	navigate?: boolean;
+}
+
+/**
+ * https://discord.com/developers/docs/topics/rpc#selectvoicechannel
+ */
+export type RPCSelectVoiceChannelResultData = RPCGetChannelResultData | null;
+
+/**
+ * https://discord.com/developers/docs/topics/rpc#getselectedvoicechannel
+ */
+export type RPCGetSelectedVoiceChannelResultData = RPCGetChannelResultData | null;
+
+/**
+ * https://discord.com/developers/docs/topics/rpc#getselectedvoicechannel
+ */
+export interface RPCGetSelectedVoiceChannelArgs {}
+
+/**
+ * @unstable
+ */
+export type RPCGetUserResultData = APIUser;
+/**
+ * @unstable
+ */
+export interface RPCGetUserArgs {
+	/**
+	 * @unstable Id of the user to get
+	 */
+	id: Snowflake;
+}
+
+/**
+ * https://discord.com/developers/docs/topics/rpc#getvoicesettings-get-voice-settings-response-structure
+ */
+export interface RPCGetVoiceSettingsResultData {
+	/**
+	 * Input settings
+	 */
+	input: RPCVoiceSettingsInput;
+	/**
+	 * Output settings
+	 */
+	output: RPCVoiceSettingsOutput;
+	/**
+	 * Voice mode settings
+	 */
+	mode: RPCVoiceSettingsMode;
+	/**
+	 * State of automatic gain control
+	 */
+	automatic_gain_control: boolean;
+	/**
+	 * State of echo cancellation
+	 */
+	echo_cancellation: boolean;
+	/**
+	 * State of noise suppression
+	 */
+	noise_suppression: boolean;
+	/**
+	 * State of voice quality of service
+	 */
+	qos: boolean;
+	/**
+	 * State of silence warning notice
+	 */
+	silence_warning: boolean;
+	/**
+	 * State of self-deafen
+	 */
+	deaf: boolean;
+	/**
+	 * State of self-mute
+	 */
+	mute: boolean;
+}
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface RPCGetVoiceSettingsArgs {}
+
+/**
+ * Returns the {@link https://discord.com/developers/docs/topics/rpc#getchannel | Get Channel} response, or `null` if none.
+ */
+export type RPCSelectTextChannelResultData = RPCGetChannelResultData | null;
+/**
+ * https://discord.com/developers/docs/topics/rpc#selecttextchannel-select-text-channel-argument-structure
+ */
+export interface RPCSelectTextChannelArgs {
+	/**
+	 * Channel id to join (or `null` to leave)
+	 */
+	channel_id: Snowflake | null;
+	/**
+	 * Asynchronously join channel with time to wait before timing out
+	 */
+	timeout?: number;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface RPCSetActivityResultData {}
+/**
+ * https://discord.com/developers/docs/topics/rpc#setactivity-set-activity-argument-structure
+ */
+export interface RPCSetActivityArgs {
+	/**
+	 * The application's process id
+	 */
+	pid: number;
+	/**
+	 * The rich presence to assign to the user
+	 */
+	activity?: Partial<
```

Comment:
```suggestion
	activity?: Partial<Omit<GatewayActivity, 'created_at' | 'id'>>
```
Also, the `type` can be narrowed further but not sure if we want to do that
![image](https://github.com/user-attachments/assets/3e0f6252-0fbc-441b-9448-046670079de9)